core/include/gnuradio-4.0/Port.hpp

```diff
@@ -163,6 +159,99 @@ static_assert(!is_stream_buffer_attribute<DefaultTagBuffer>::value);
+    Annotated<float, "signal min,", Doc<"minimum expected signal value">>                                            signal_min = std::numeric_limits<float>::lowest();
+    Annotated<float, "signal max,", Doc<"maximum expected signal value">>                                            signal_max = std::numeric_limits<float>::max();
+
+    // controls automatic (if set) or manual update of above parameters
+    std::set<std::string, std::less<>> auto_update{ "sample_rate", "signal_name", "signal_quantity", "signal_unit", "signal_min", "signal_max" };
+
+    PortMetaInfo() noexcept(true) : PortMetaInfo({}) {}
+
+    explicit PortMetaInfo(std::initializer_list<std::pair<const std::string, pmtv::pmt>> initMetaInfo) noexcept(true) : PortMetaInfo(property_map{ initMetaInfo.begin(), initMetaInfo.end() }) {}
+
+    explicit PortMetaInfo(const property_map &metaInfo) noexcept(true) { update<true>(metaInfo); }
+
+    void
+    reset() {
+        auto_update = { "sample_rate", "signal_name", "signal_quantity", "signal_unit", "signal_min", "signal_max" };
+    }
+
+    template<bool isNoexcept = false>
+    void
+    update(const property_map &metaInfo) noexcept(isNoexcept) {
+        if (metaInfo.empty()) {
+            return;
+        }
+
+        auto updateValue = [&metaInfo](const std::string &key, auto &member) {
+            if (!metaInfo.contains(key)) {
+                return;
+            }
+            const auto &value = metaInfo.at(key);
+            using T           = std::decay_t<decltype(member.value)>;
+            if (std::holds_alternative<T>(value)) {
+                member = std::get<T>(value);
+            } else {
+                throw gr::exception("invalid-argument: incorrect type for " + key);
+            }
+        };
+
+        for (const auto &key : auto_update) {
+            if (key == "sample_rate") {
+                updateValue(key, sample_rate);
+            } else if (key == "signal_name") {
+                updateValue(key, signal_name);
+            } else if (key == "signal_quantity") {
+                updateValue(key, signal_quantity);
+            } else if (key == "signal_unit") {
+                updateValue(key, signal_unit);
+            } else if (key == "signal_min") {
+                updateValue(key, signal_min);
+            } else if (key == "signal_max") {
+                updateValue(key, signal_max);
+            }
+        }
+    }
+
+    [[nodiscard]] property_map
+    get() const noexcept {
+        property_map metaInfo;
+        metaInfo["sample_rate"]     = sample_rate;
+        metaInfo["signal_name"]     = signal_name;
+        metaInfo["signal_quantity"] = signal_quantity;
+        metaInfo["signal_unit"]     = signal_unit;
+        metaInfo["signal_min"]      = signal_min;
+        metaInfo["signal_max"]      = signal_max;
+
+        return metaInfo;
+    }
+};
+
+}
+ENABLE_REFLECTION(gr::PortMetaInfo, sample_rate, signal_name, signal_quantity,signal_unit, signal_min, signal_max)
```

Comment:
Yes, with reflection this would need to iterate through a tuple. Since these meta-data are not user-defined it's unlikely that new fields are spontaneously added. I.e. a manual access compiles a bit faster (saw this in the traces while optimising Settings).